src/utils/apiResponse.utils.ts

```diff
@@ -0,0 +1,29 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+export type ControllerResponse = {
+  isSuccess: number;
+  statusCode: number;
+  message: string;
+  data?: any;
+  error?: any;
+};
+
+export const APIResponse = {
+  create: (
+    isSuccess: number,
+    statusCode: number,
+    message?: string,
+    data?: any,
+    error?: any,
+  ): ControllerResponse => {
+    const response: ControllerResponse = { isSuccess, statusCode, message };
+    if (error) {
+      response.error = error;
+    } else if (!error && data) {
+      response.data = data;
+    } else {
+      response.data = {};
```

Comment:
PullProofAI

The `APIResponse.create` function in `apiResponse.utils.ts` has a potential logical issue. The current logic sets `response.data` to an empty object if there is no error and no data provided. This behavior might lead to unexpected results or confusion. It's advisable to handle this scenario more explicitly to avoid ambiguity.

Consider revising the logic to provide a clearer indication when no data is present without defaulting to an empty object.

```diff
-    } else {
-      response.data = {};
+    } else if (!data) {
+      response.data = null;
```


<!-- This is an auto-generated comment by PullProofAI -->